src/app.js

```diff
@@ -598,5 +600,12 @@ app.useAsync(async (err, req, res, next) => {
   return next(err);
 });
 
+/**
+ * Update the resources we know about. If the underlying data (a JSON file on
+ * S3) hasn't changed then this only involves a HEAD request so we run it on an
+ * hourly interval.
+ */
+updateResourceVersions();
+setInterval(updateResourceVersions, 60*60*1000)
 
```

Comment:
The index (and eTag) is held in memory, so each dyno is completely independent. If the dyno restarts it will grab the latest index at start time.

It's possible that requests will behave differently depending on which dyno handles the response, but this behaviour will only last up to 1 hour (eventual consistency) and only be applicable in a very specific situation, such as:
* Dataset X updated on 2023-01-01 and 2023-02-01
* Index 1 includes datasets for X up to version 2023-01-01
* Index 2 includes datasets for X up to version 2023-02-01
* Dyno 1 uses index 1 (because it hasn't yet updated), Dyno 2 uses index 2 
* Request for X@2023-01-05 will return different versions: dyno 1 will return the latest version, which will be 2023-02-01. Dyno 2 will return the version from 2023-01-01.

Reply:
Seems ok for now. Maybe this can use the generic key-value storage in #739 once that's merged?

Reply:
I'm not sure the key-value storage is the correct place for this index to be shared, but yes, sharing it across instances would be a good future improvement.